### PR TITLE
Add option to save benchmark results to csv

### PR DESCRIPTION
**Pull request by kgajdamo**
The csv files will be saved to the benchmark/results folder.

### CODE REVIEW

File: benchmark/utils/utils.py [hunk 1]
```diff
@@ -116,3 +119,30 @@ def get_split_masks(data, dataset_name):
...
+    csv_data['MODEL'].append(model_name)
+    csv_data['DATASET'].append(dataset_name)
+    csv_data['CONFIG'].append(config)
+    csv_data['SPARSE'].append(use_sparse_tensor)
+
+
+def write_to_csv(csv_data, training=False):
+    results_path = osp.join(osp.dirname(osp.realpath(__file__)), '../results/')
+    if not osp.exists(results_path):
+        os.makedirs(results_path)
```
Comment (rusty1s) [inline benchmark/utils/utils.py:142]:
```suggestion
    os.makedirs(results_path, exist_ok=True)
```

File: benchmark/utils/utils.py [hunk 2]
```diff
@@ -116,3 +119,30 @@ def get_split_masks(data, dataset_name):
         val_mask = data.val_mask
         test_mask = data.test_mask
     return train_mask, val_mask, test_mask
+
+
+def save_benchmark_data(csv_data, batch_size, layers, num_neighbors,
+                        hidden_channels, total_time, model_name, dataset_name,
+                        use_sparse_tensor):
+    config = f'Batch size={batch_size}, ' \
+             f'Layers amount={layers}, ' \
```
Comment (rusty1s) [inline benchmark/utils/utils.py:128]:
```suggestion
             f'#Layers={layers}, ' \
```

File: benchmark/utils/utils.py [hunk 3]
```diff
@@ -116,3 +119,30 @@ def get_split_masks(data, dataset_name):
...
+             f'Num_neighbors={num_neighbors}, ' \
+             f'Hidden features size={hidden_channels}'
```
Comment (rusty1s) [inline benchmark/utils/utils.py:130]:
```suggestion
             f'#Hidden features={hidden_channels}'
```

File: benchmark/utils/utils.py [hunk 4]
```diff
@@ -116,3 +119,30 @@ def get_split_masks(data, dataset_name):
...
         test_mask = data.test_mask
     return train_mask, val_mask, test_mask
+
+
+def save_benchmark_data(csv_data, batch_size, layers, num_neighbors,
+                        hidden_channels, total_time, model_name, dataset_name,
+                        use_sparse_tensor):
+    config = f'Batch size={batch_size}, ' \
+             f'Layers amount={layers}, ' \
+             f'Num_neighbors={num_neighbors}, ' \
```
Comment (rusty1s) [inline benchmark/utils/utils.py:129]:
```suggestion
             f'#Neighbors={num_neighbors}, ' \
```

File: benchmark/inference/inference_benchmark.py [hunk 5]
```diff
@@ -225,6 +229,15 @@ def run(args: argparse.ArgumentParser):
                         print(f'Throughput: {throughput:.3f} samples/s')
                         print(f'Latency: {latency:.3f} ms')
 
+                        if args.write_csv:
```
Comment (rusty1s) [inline benchmark/inference/inference_benchmark.py:232]: I think there is no real reason to gate this behind `if args.write_csv`. I would just gate the final `write_to_csv` call.

**Review by rusty1s — APPROVED** [review]

File: benchmark/utils/utils.py (hunk 1 above)
Comment (kgajdamo) [reply benchmark/utils/utils.py]: Good idea, thanks!

File: benchmark/utils/utils.py (hunk 2 above)
Comment (kgajdamo) [reply benchmark/utils/utils.py]: Done.

File: benchmark/utils/utils.py (hunk 4 above)
Comment (kgajdamo) [reply benchmark/utils/utils.py]: Done. Thank You for the code review :)

File: benchmark/inference/inference_benchmark.py (hunk 5 above)
Comment (kgajdamo) [reply benchmark/inference/inference_benchmark.py]: Done.

File: benchmark/training/training_benchmark.py [hunk 6]
```diff
@@ -249,6 +258,15 @@ def run(args: argparse.ArgumentParser):
                         print(f'Throughput: {throughput:.3f} samples/s')
                         print(f'Latency: {latency:.3f} ms')
 
+                        if args.write_csv:
+                            save_benchmark_data(csv_data, batch_size, layers,
+                                                num_neighbors, hidden_channels,
+                                                total_time, model_name,
+                                                dataset_name,
+                                                args.use_sparse_tensor)
```
Comment (rusty1s) [inline benchmark/training/training_benchmark.py:266]:
```suggestion
                        save_benchmark_data(csv_data, batch_size, layers,
                                            num_neighbors, hidden_channels,
                                            total_time, model_name,
                                            dataset_name,
                                            args.use_sparse_tensor)
```

**Review by rusty1s — APPROVED** [review]

**Pull request merged**